network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ConnectablePositionImpl.java

```diff
@@ -68,9 +103,105 @@ public Direction getDirection() {
+            branch.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setLabel(Objects.requireNonNull(name));
+                }
+            });
+        }
+
+        private void updateTransformerLabelResource(ThreeWindingsTransformerImpl windingsTransformer, String name) {
+            windingsTransformer.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.THREE) {
+                    res.getAttributes().getPosition3().setLabel(Objects.requireNonNull(name));
+                }
+            });
+        }
+
+        private void updateInjectionOrderResource(AbstractInjectionImpl<?, ?> injection, int order) {
+            injection.updateResource(res -> res.getAttributes().getPosition().setOrder(order));
+        }
+
+        private void updateBranchOrderResource(AbstractBranchImpl<?, ?> branch, int order) {
+            branch.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setOrder(order);
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setOrder(order);
+                }
+            });
+        }
+
+        private void updateTransformerOrderResource(ThreeWindingsTransformerImpl windingsTransformer, int order) {
+            windingsTransformer.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setOrder(order);
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setOrder(order);
+                } else if (getSide() == ThreeSides.THREE) {
+                    res.getAttributes().getPosition3().setOrder(order);
+                }
+            });
+        }
+
+        private void updateInjectionDirectionResource(AbstractInjectionImpl<?, ?> injection, Direction direction) {
+            injection.updateResource(res -> res.getAttributes().getPosition()
+                    .setDirection(Direction.valueOf(Objects.requireNonNull(direction).name())));
+        }
+
+        private void updateBranchDirectionResource(AbstractBranchImpl<?, ?> branch, Direction direction) {
+            branch.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setDirection(Objects.requireNonNull(direction));
```

Comment:
I think Objects.requireNonNull(direction) could be add at the beginning of th method